### PR TITLE
GCB-235: Configure bubble resize in configuration.json

### PR DESCRIPTION
**Pull request by michallaskowski**

### CODE REVIEW

File: GrandCentralBoard/Widgets/Bonus/Scene/Bubble.swift [hunk 1]
```diff
@@ -9,15 +9,17 @@ class Bubble: SKSpriteNode, BubbleScalingAnimatorDelegate {
...
+    private let bubbleResizeDuration: NSTimeInterval
     private lazy var scalingAnimator: BubbleScalingAnimator = {
-        let animator = BubbleScalingAnimator(spriteNode: self)
+        let animator = BubbleScalingAnimator(spriteNode: self, bubbleResizeDuration: self.bubbleResizeDuration)
         animator.delegate = self
         return animator
     }()
     
-    init(bubbleViewModel: BubbleViewModel) {
+    init(bubbleViewModel: BubbleViewModel, bubbleResizeDuration: NSTimeInterval) {
```
Comment (Augustyniak) [inline GrandCentralBoard/Widgets/Bonus/Scene/Bubble.swift]: `bubbleResizeDuration` should be (IMO) a part of the `bubbleViewModel` - view models configures the view and `bubbleResizeDuration` is a configuration parameter.

File: GrandCentralBoard/Widgets/Bonus/Scene/BubbleScalingAnimator.swift [hunk 2]
```diff
@@ -24,9 +24,12 @@ final class BubbleScalingAnimator {
...
     private var scaleDownTimer: NSTimer?
     weak var delegate: BubbleScalingAnimatorDelegate?
-    
-    init(spriteNode: SKSpriteNode) {
+
+    private let bubbleResizeDuration: NSTimeInterval
+
+    init(spriteNode: SKSpriteNode, bubbleResizeDuration: NSTimeInterval) {
         self.spriteNode = spriteNode
+        self.bubbleResizeDuration = bubbleResizeDuration
```
Comment (Augustyniak) [inline GrandCentralBoard/Widgets/Bonus/Scene/BubbleScalingAnimator.swift]: Can we validate whether this number is greater than 0?

File: GrandCentralBoard/Widgets/Bonus/BonusWidgetBuilder.swift [hunk 3]
```diff
@@ -34,7 +45,12 @@ struct BonusWidgetSettings: Decodable {
     let bubbleResizeDuration: NSTimeInterval
 
     static func decode(jsonObject: AnyObject) throws -> BonusWidgetSettings {
-        return try BonusWidgetSettings(accessToken: jsonObject => "accessToken",
+        let settings = try BonusWidgetSettings(accessToken: jsonObject => "accessToken",
                                        bubbleResizeDuration: jsonObject => "bubbleResizeDuration")
+
+        guard settings.bubbleResizeDuration > 0 else {
+            throw BonusWidgetSettingsError.BubbleResizeDurationInvalid(settings.bubbleResizeDuration)
```
Comment (Augustyniak) [inline GrandCentralBoard/Widgets/Bonus/BonusWidgetBuilder.swift]: 👍

File: GrandCentralBoard/Widgets/Bonus/BonusWidgetBuilder.swift [hunk 4]
```diff
@@ -26,6 +26,17 @@ final class BonusWidgetBuilder : WidgetBuilding {
     }
 }
 
+enum BonusWidgetSettingsError: ErrorType, HavingMessage {
+    case BubbleResizeDurationInvalid(NSTimeInterval)
+
+    var message: String {
+        switch self {
+        case .BubbleResizeDurationInvalid(let duration):
+            return "BubbleResizeDuration of \(duration) is invalid."
```
Comment (Augustyniak) [inline GrandCentralBoard/Widgets/Bonus/BonusWidgetBuilder.swift]: I'm not sure whether this string literal is valid.

**Comment by Augustyniak** [comment]
I don't have more comments. Please answer my doubts. 🎾

**Comment by michallaskowski** [comment]
I checked and corrected. Thanks.

**Pull request merged**